src/tests/xkey/test12.vtc

```diff
@@ -0,0 +1,110 @@
+varnishtest "Test xkey bytes gauge"
+
+server s1 {
+	rxreq
+	txresp -hdr "xkey: asdf"
+	rxreq
+	txresp -hdr "xkey: fdsa"
+	rxreq
+	txresp -hdr "xkey: asdf"
+} -start
+
+varnish v1 -vcl+backend {
+	import xkey from "${vmod_builddir}/.libs/libvmod_xkey.so";
+
+	sub vcl_recv {
+		if (req.http.xkey-purge) {
+			if (xkey.purge(req.http.xkey-purge) != 0) {
+				return (synth(200, "Purged"));
+			} else {
+				return (synth(404, "No key"));
+			}
+		}
+	}
+
+	sub vcl_backend_response {
+		set beresp.ttl = 60s;
+		set beresp.grace = 0s;
+		set beresp.keep = 0s;
+	}
+
+	sub vcl_synth {
+		set resp.http.reason = resp.reason;
+	}
+} -start
+
+# g_hashhead_bytes  = 104 bytes per xkey
+#
```

Comment:
I don't think the detail is needed here, as it could change, and you don't want to duplicate the changes. You can however point at the header containing the info

Reply:
The reason we added this detail is that it could change, which would break the test. This detail should help figure out how and why the test is broken and how to update it if the break is valid.

Reply:
but what is going to happen is really that this specific test will break and it'll be pretty obvious that the struct size changed, we'll just re-ask `gcc`/`gdb` for the new size, update the test and be on our way.
The actual internal struct matters very little, and you can still reference the header file anyway

Reply:
What we care about in the test is that we increment by a "headhead unit", however many bytes that happens to be. That's why we have the number of bytes for the structs in the test.

I agree that all of the details about the contents of each struct is not necessary. We were using that for our own research while working on this.